### ISSUE TITLE
The ‘module.hot.accept’ method reports an error when I using asynchronous modules

### ISSUE THREAD

**Issue by sampsonli** [webpack/webpack#11728]
# Bug report

**What is the current behavior?**
>
>Uncaught TypeError: Cannot read property 'call' of undefined
>  at __webpack_require__ (bootstrap:21)
> at fn (hot module replacement:60)
> at Module../src/routes/demo/index.js (app.js:351)
> at __webpack_require__ (bootstrap:21)
> at fn (hot module replacement:60)

**If the current behavior is a bug, please provide the steps to reproduce.**
~~~javascript
// this code report error!
const Home = load(() => import('./pages/home'));
module.hot.accept('./pages/home', () => {
  console.log('hello');
});

~~~

but below code is OK!
~~~javascript

import Home from './pages/home';
module.hot.accept('./pages/home', () => {
  console.log('hello');
});

~~~
~~~


<!-- A great way to do this is to provide your configuration via a GitHub repository -->
<!-- The most helpful is a minimal reproduction with instructions on how to reproduce -->
<!-- Repositories with too many files or large `webpack.config.js` files are not suitable -->
<!-- Please only add small code snippets directly into this issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader, please create an issue on the loader/plugin repository instead -->

**What is the expected behavior?**
I'd like async module could  reload before execute myself logic



<!-- "It should work" is not a helpful explanation -->
<!-- Explain exactly how it should behave -->

**Other relevant information:**
webpack version:  5.1.3
Node.js version: 12.11.1
Operating System:  win10
Additional tools:

**Comment by evilebottnawi** [comment]
Please fill out all files in the issue template, I will reopen this after your fixed

**Comment by sampsonli** [comment]
The problem has been solved in v5.3.0
